sig-wg-lifecycle.md

```diff
@@ -56,13 +56,13 @@ Out of scope for this document: [subproject] creation.
 ### Communicate:
 Each one of these has a linked canonical source guideline from set up to moderation and your role and responsibilities for each. We are all responsible for enforcing our [code of conduct].
 - [ ] Read [moderation.md] and understand your role in keeping our community safe
-- [ ] Create your mailing lists - One for your members and another for your chairs/leads
-- Either [GoogleGroups] OR [discuss.kubernetes.io]
```

Comment:
removed discuss.kubernetes.io as we want to streamline mailing list creation. discussed in https://kubernetes.slack.com/archives/C1TU9EB9S/p1593624758123000 with @parispittman and @castrojo